core/src/apps/cardano/layout.py

```diff
@@ -331,6 +357,57 @@ async def show_warning_tx_output_contains_tokens(ctx: wire.Context) -> None:
     )
 
 
+async def show_warning_tx_output_contains_datum_hash(
+    ctx: wire.Context, datum_hash: bytes
+) -> None:
+    await confirm_properties(
+        ctx,
+        "confirm_datum_hash",
+        title="Confirm transaction",
+        props=[
+            (
+                "The following\ntransaction output\ncontains datum hash:",
```

Comment:
I'm not sure why we used the new lines here, but it seems it's not needed. The lines will be split automatically and correctly by the underlying code. Also in the warnings below.

Reply:
Cool that you noticed this; removed.